android/src/main/java/com/lwansbrough/RCTCamera/RCTCameraViewFinder.java

```diff
@@ -300,63 +310,91 @@ public void onPreviewFrame(byte[] data, Camera camera) {
-            byte[] rotated = new byte[imageData.length];
-            for (int y = 0; y < height; y++) {
-              for (int x = 0; x < width; x++) {
-                rotated[x * height + height - y - 1] = imageData[x + y * width];
-              }
+            // find rotation that will be used for Mutable Image
+            int deviceRotation = settings.getActualDeviceOrientation();
+            int rotationIndex = 0;
+            switch(deviceRotation){
+                case 0: // portrait
+                    rotationIndex = 6;
+                    break;
+                case 1: // landscape left
+                    rotationIndex = 1;
+                    break;
+                case 2: // portrait upside down
+
+                    break;
+                case 3: // landscape right
+                    rotationIndex = 3;
+                    break;
             }
-            imageData = rotated;
-        }
 
-        @Override
-        protected Void doInBackground(Void... ignored) {
-            if (isCancelled()) {
+            // rotate
+            MutableImage mutableImage = new MutableImage(this.imageData);
+            try{
+                mutableImage.rotate(rotationIndex);
+            } catch (MutableImage.ImageMutationFailedException e) {
+                android.util.Log.e("RCTCamera","Rotate temp image",e);
                 return null;
             }
 
+            Bitmap bitmap = mutableImage.getCurrentRepresentation();
+            int width; int height;
+
+            // if using viewfinder, crop search area
+            if ( settings.barcodeFinderVisible() && settings.barcodeFinderHeight() != 0 && settings.barcodeFinderWidth() != 0) {
+
+                // Get actual size based on %
+                width = (int) (bitmap.getWidth() * settings.barcodeFinderWidth() );
+                height = (int) (bitmap.getHeight() * settings.barcodeFinderHeight() );
+                // find center - new position
+                int x = (bitmap.getWidth() / 2) - (width / 2);
+                int y = (bitmap.getHeight() / 2) - (height / 2);
+                // our small image
+                bitmap = Bitmap.createBitmap(bitmap, x, y, width, height);
+            }
+            else {
+                // fullscreen
+                width = bitmap.getWidth();
+                height = bitmap.getHeight();
+            }
+
+            // get dataarray for actual area to scan
+            int[] intArray = new int[width*height];
+            bitmap.getPixels(intArray, 0, width, 0, 0, width, height);
+
             try {
-                // rotate for zxing if orientation is portrait
-                Result result = getBarcodeAnyOrientation();
-                if (result == null){
-                    throw new Exception();
-                }
+                LuminanceSource source = new RGBLuminanceSource(width, height, intArray);
+                BinaryBitmap bbitmap = new BinaryBitmap(new HybridBinarizer(source));
```

Comment:
Why not continue using YUV format and [PlanarYUVLuminanceSource][2] and switching to  [`RGBLuminanceSource`][1]? I am trying to understand the reason as I am also beginner in this field. Thanks a lot

[2]: https://zxing.github.io/zxing/apidocs/com/google/zxing/PlanarYUVLuminanceSource.html

[1]: https://zxing.github.io/zxing/apidocs/com/google/zxing/RGBLuminanceSource.html